streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -880,6 +896,9 @@ private void initializeChangelogs(final Map<TaskId, Task> tasks,
     }
 
     private void addChangelogsToRestoreConsumer(final Set<TopicPartition> partitions) {
+        if (partitions.isEmpty())
```

Comment:
This and the line 929 below is to reduce verbose logging lines of KafkaConsumer when it calls `unsubscribe`: if the partitions set is empty, while the current assignment is also empty, we should not proceed at all.

Reply:
> while the current assignment is also empty

How do we know this? We did not get the assignment yet? Should this go after
```
final Set<TopicPartition> assignment = new HashSet<>(restoreConsumer.assignment());

if (partitions.isEmpty() && assignment.isEmpty()) {
  return;
}
```

Reply:
Yes that's right, but as long as `partitions.isEmpty()` then we could actually skip the whole func, so I made it applied at larger scope.